arch/arm/core/aarch32/swap_helper.S

```diff
@@ -556,8 +556,35 @@ _stack_frame_endif:
 
 _oops:
     push {r0, lr}
+#if defined(CONFIG_EXTRA_EXCEPTION_INFO)
+#if defined(CONFIG_ARMV7_M_ARMV8_M_MAINLINE)
+    /* First push an unnecessary extra register so
+     * that stack pointer will be 8-byte aligned
+     * after pushing callee saved onto the stack
+     */
+    push {r2}
+
+    /* Build _callee_saved_t. To match the struct
+     * definition we push the psp & then r11-r4
+     */
+    mrs r1, PSP
+    push {r1}
```

Comment:
You can just do `push {r1, r2}` here.

> The registers are stored in sequence, the lowest-numbered register to the lowest
memory address (start_address), through to the highest-numbered register to the
highest memory address (end_address).

Note that we do the same here:
https://github.com/zephyrproject-rtos/zephyr/blob/30cd0d4338601ffd39d8f5d1a992cb4f090ced6e/arch/arm/core/aarch32/cortex_m/fault_s.S#L83-L100